File: src/my_project/easy_problems/from201to250/minimum_distance_to_target.py
from typing import List 

class Solution:
    def getMinDistance(self, nums: List[int], target: int, start: int) -> int:
        index_list  = list()
        len_nums = len(nums)
        for i in range(len_nums):
            if nums[i] == target:
                index_list.append(i)

        
        
        index_list.sort(key= lambda x: abs(x-start) )
            
        return abs(index_list[0] - start)
    
    
    
solution = Solution()

print(solution.getMinDistance( nums = [1,2,3,4,5], target = 5, start = 3))

print(solution.getMinDistance( nums = [1], target = 1, start = 0))

print(solution.getMinDistance( nums = [1,1,1,1,1,1,1,1,1,1], target = 1, start = 0))

File: my_project_typescript/easy_problems_typescript/from101to150/minimumDistanceToTarget.ts
function getMinDistance(nums: number[], target: number, start: number): number {
    let indexList: Array<number> = []
    let lenNums: number = nums.length

    for (let i = 0; i < lenNums; i++){
        if (nums[i] === target){
            indexList.push(i)
        }
    }

    indexList = indexList.sort( (a,b) => Math.abs(a-start) <= Math.abs(b-start) ?-1:1 )
            
    return Math.abs(indexList[0] - start)
};


console.log(getMinDistance([1,2,3,4,5],  5,  3))

console.log(getMinDistance([1], 1,  0))

console.log(getMinDistance( [1,1,1,1,1,1,1,1,1,1],  1,  0))
